bolt-simple-taproot.md

```diff
@@ -377,348 +352,276 @@ The Context column decodes as follows:
+- MUST use the generated secret nonce and the calculated aggregate nonce to construct a
+  `musig2` partial signature for the sender's remote commitment using the `Sign` algorithm from `bip-musig2`.
+- MUST include the partial signature and the public counterpart of the generated nonce in
+  the `partial_signature_with_nonce` field.
+- MUST generate a second unique nonce to send in the `musig2_nonce` field. This nonce MUST be used to sign the sender's
+  local commitment using the `Sign` algorithm from `bip-musig2`.
 
-#### `closing_signed` Extensions
+The recipient:
 
-Similar to the `shutdown` message, we add a new TLV to the `closing_signed`
-message's existing `tlv_stream`:
+- MUST fail the channel if `signature` is non-zero.
+- MUST fail the channel if `partial_signature_with_nonce` is absent.
+- MUST fail the channel if `musig2_nonce` is absent.
+- MUST compute the aggregate nonce from:
+  - the `musig2_nonce` field the recipient previously sent in the latest of `channel_ready`,
+    `channel_reestablish`, `revoke_and_ack`, or `closing_signed`
+  - the `public_nonce` included as part of the `partial_signature_with_nonce` field
+- MUST verify the `partial_signature_with_nonce` field using the `PartialSigVerifyInternal`
+  algorithm of `bip-musig2`:
+  - if the partial signature is invalid, MUST fail the channel
+- MUST store `musig2_nonce` until they, in turn, send _their_ `closing_signed` message.
+  - SHOULD only store `musig2_nonce` in RAM, and not persist it to disk.
 
-1. `tlv_stream`: `closing_signed_tlvs`
-2. types:
-    1. type: 2 (`musig2_pubnonce`)
-    2: data:
-        * [`66*byte`:`nonces`]
+### Channel Operation
 
-Both sides **MUST** provide this new TLV field.
+`n_a_L`: Alice's local secret nonce
 
-As a fresh `musig2_pubnonce` is required to sign another fee proposal, with
-each new signature, both sides send a new nonce that'll be used for the _next_
-signature.
+`n_B_R`: Bob's remote public nonce
 
-### Channel Operation
+![](taproot_channel_operation.jpg)
 
 #### `commitment_signed` Extensions
 
 A new TLV stream is added to the `commitment_signed` message:
 
-1. type: 132 (`commitment_signed`)
-2. data:
-    * ...
-    * [`commitment_signed_tlvs`:`tlvs`]
-
 1. `tlv_stream`: `commitment_signed_tlvs`
 2. types:
-    1. type: 2 (`signing_musig2_pubnonce`)
-    2: data:
-        * [`66*byte`:`nonces`]
-
-The main commitment `signature` attached in the message is now a `bip-musig2`
-_partial_ schnorr signature. Using the `secnonce` generated upon connection
-re-establishment or funding confirmation, the partial signature is generated
-with the `Sign` algorithm specified in the `musig2` BIP.
-
-When creating a new signature, the sender attaches their _signing_ nonce which
-allows the receiver to generate an additional commitment.
+   1. type: 2 (`partial_signature_with_nonce`)
+   2. data:
+      * [`98*byte`: `partial_signature || public_nonce`]
```

Comment:
I think _this_ change to the commitment dance is much more compelling that the initial funding changes. 

Before it was:
  * both sides exchange signing+verification nonces at the start of the connection
  * when you sign, you send your next signing nonce (you'll use to sign state N+1)
  * when they revoke, they send their next verification nonce (you'll use to sign state N+1)

With this it's:
  * both sides exchange a verification/local nonce at the start 
  * you sign sign, you send your signing nonce (lets them verify it)
  * when they revoke, they send their next verification nonce

Reply:
Typing it out then reading it back, doesn't seem as striking, since the revoke is more or less the same, and you're still sending a nonce w/ the sig, just that it's the next/current nonce instead.

Reply:
I think one nice part about sending the nonce along w/ the sig at the final step is: the message itself can be passed as aux input to the nonce generation function, binding the nonce to the commitment transaction itself (see prior point about being able to retransmit w/ diff HTLCs).